client/src/components/searchBar.jsx

```diff
@@ -29,23 +42,47 @@ const searchBar = (props) => {
     setAnchorEl(null)
   }
 
-
   const handleSubmit = (e) => {
-    e.preventDefault()
+    e.preventDefault();
     if (searchInput.length > 0) {
-      props.getStockData(searchInput, searchScope, 'search')
-      e.target.reset()
+      // Update search history
+      const newSearchHistory = [
+        ...searchHistory,
+        { searchInput, searchScope },
+      ];
+      setSearchHistory(newSearchHistory);
+      console.log(newSearchHistory)
+      localStorage.setItem('searchHistory', JSON.stringify(newSearchHistory));
+
+      props.getStockData(searchInput, searchScope, 'search');
+      e.target.reset();
       navigate(`/searchContent`);
     }
+  };
+
+  const handleSearchFocus = () => {
+    const storedSearchHistory = localStorage.getItem('searchHistory');
+    console.log('focused', storedSearchHistory)
+    setShowSearchHistory(true);
+
+    if (storedSearchHistory) {
+      setSearchHistory(JSON.parse(storedSearchHistory));
```

Comment:
Parse is very finicky meaning even the smallest of typos will break your whole application if uncaught. 

Catch your errors! for JSON parse it needs to be a try catch https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch